### DISCUSSION TITLE
Auto keyboard backlight setting for Lenovo Thinkpad T16

### DISCUSSION THREAD

**Discussion by ghost** [AutoDarkMode/Windows-Auto-Night-Mode#718] [Custom scripts]
I created a powershell script to turn on/off the keyboard backlighting for Lenovo Thinkpad T16 (2022 model).  it doesn't have auto-off settings in BIOS/app.  This was disseminated from other tools that control lighting on Thinkpads.  NO WARRANTY GIVEN OR IMPLIED.  This script relies on the Lenovo utilties libraries that are installed as default in the Lenovo Window 10 factory OS image.  Should work in Win 11 too as long as the thinkpad tools are compatible.   Script is calling the PS scripts using powershell(x86) (32bit) since it does not work with 64bit powershell - I guess the libraries are 32bit.  Prior to using these scripts, I opened 32bit powershell(x86) as admin and enabled the script execution bypass (use the Set-ExecutionPolicy Bypass command).  You can test these manually from powershell(x86) by executing .\keylow.ps1 and .\keyoff.ps1.  Hope this is useful to the Thinkpad users - not sure why Lenovo couldn't have just added auto-off feature for the keyboard backliight like every other manufacturer.
[Lenovo_T16_keyboard_toggle.zip](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/files/11360519/Lenovo_T16_keyboard_toggle.zip)

**Comment by TurboJens** [comment]
Cool, any idea of how to automate this so that it will illuminate when keyboard/touchpad is used and turn of after a few seconds of inactivity? So it would behave like every other manufacturer do?
